contracts/PortfolioLib.sol

```diff
@@ -249,6 +249,9 @@ function getPoolReserves(PortfolioPool memory self)
     pure
     returns (uint128 reserveAsset, uint128 reserveQuote)
 {
+    // Check if -`self.liquidity` fits within an int128 by checking if it's
+    // greater than the minimum negative value of an int128.
+    if (self.liquidity > 170141183460469231731687303715884105728) revert();
```

Comment:
I'll note that this function still reverts if `self.liquidity == 170141183460469231731687303715884105728`, because the `int128` cast underflows, resulting in an attempt to negate `-2^127` as an `int128`, an operation which is checked. While you could restructure the code to allow the maximal input, it may not be worth the trouble. I think the most straightforward way to do it would be doing `int128(-int256(uint256(self.liquidity)))` instead of the current cast-and-negation, although there might be a more elegant option.

Reply:
Fixed!